### DISCUSSION TITLE
Send more game parameters in Battlesnake API requests

### DISCUSSION THREAD

**Discussion by bvanvugt** [BattlesnakeOfficial/feedback#75] [Feature Requests]
From [penelope_zone on twitter](https://twitter.com/penelope_zone/status/1409146170044264461):

> hey @playbattlesnake would you be able to send the amount of damage hazard sauce does, as well as how often food spawns, how often hazard spawns, etc, with the request payload? I think that'd really help us simulators :D

It'd be great if game requests included the following information:
* [All] Food Spawn Rate
* [Royale] Hazard Spawn Rate
* [Royale] Hazard Damage Rate

**Comment by bvanvugt** [comment]
For current reference, these values are currently hardcoded in their respective game modes:

Food Spawn Rate: 15% chance each turn
Hazard Spawn Rate: every 20 turns
Hazard Damage: -15 HP (in addition to cost of moving)

**Reply by xtagon** [reply]
They're hard-coded, but there are different variations of game modes with different values, aren't there? For example yesterday we Snake Pit Live had a few variants of hazard spiceyness.

I would love to see challenge criteria names in the API request. Presently the workaround for challenges is to change the snake's URL or parameters before attempting a challenge because they all have different goals.

**Reply by bvanvugt** [reply]
Yeah, that's true. The SPL game variants adjust these values with reckless abandon, which is notably unfair to the snakes but also fun to watch.

I like the idea of passing challenge info - we'd have to think had about how though. Maybe those are sent as string sentences to start?

**Comment by xtagon** [comment]
Related: https://github.com/BattlesnakeOfficial/feedback/discussions/88

**Reply by coreyja** [reply]
#88 item was likely mis-informed, these come back in the `games` API endpoint that is used to download games, but I do NOT think it is actually getting sent to the snakes like I originally thought

**Comment by bvanvugt** [comment]
This is becoming increasingly important as we explore Royale mode competitively, prioritizing it as something we'd like to address short-term.
